src/h5web/vis-packs/core/rgb/RgbVis.tsx

```diff
@@ -3,39 +3,60 @@ import styles from '../heatmap/HeatmapVis.module.css';
 import PanZoomMesh from '../shared/PanZoomMesh';
 import VisCanvas from '../shared/VisCanvas';
 import type { Layout } from '../heatmap/models';
-import { DataTexture, RGBFormat, UnsignedByteType } from 'three';
+import { DataTexture, FloatType, RGBFormat, UnsignedByteType } from 'three';
 import VisMesh from '../shared/VisMesh';
+import { flipLastDimension } from './utils';
+import { ImageType } from './models';
+import { DTypeClass, NumericType } from '../../../providers/models';
 
 interface Props {
   value: number[];
   dims: number[];
+  dtype: NumericType;
```

Comment:
Creating the texture is the responsibility of the vis component, so there is no way around passing some sort of type information. That being said, we could just pass a boolean, like `withFloats` so that our "HDF5" models don't leak into the vis components.